absl/strings/escaping_test.cc

```diff
@@ -278,6 +278,98 @@ TEST_F(CUnescapeTest, UnescapesMultipleUnicodeNulls) {
                    "\0", 5), result_string_);
 }
 
+TEST(CUnescape, NonNullTerminatedOctal1) {
+  absl::string_view reference("\\111", 2);
+
+  std::string unescaped_str;
+  EXPECT_TRUE(absl::CUnescape(reference, &unescaped_str));
+
+  EXPECT_EQ(unescaped_str, "\1");
+}
+
+TEST(CUnescape, NonNullTerminatedOctal2) {
+  absl::string_view reference("\\1z1", 3);
```

Comment:
Does this follow a different code path than `absl::string_view reference("\\1z");` would?  If not, we should choose the simpler test input.  We could then rename it `InvalidOctal` and put it with the other similar tests below.

Reply:
They kind of follow the same code path, but are designed to catch different "off by one" mistakes.

The relevant code piece is this one:
escaping.cc
```
 144           if (p < last_byte && is_octal_digit(p[1])) ch = ch * 8 + *++p - '0';                                          
 145           if (p < last_byte && is_octal_digit(p[1]))                                                                                                                        
 146             ch = ch * 8 + *++p - '0';      // now points at last digit    
```

While the first test catches things like:

```
 -144           if (p < last_byte && is_octal_digit(p[1])) ch = ch * 8 + *++p - '0';
 +144           if (p <= last_byte && is_octal_digit(p[1])) ch = ch * 8 + *++p - '0';                                          
 145           if (p < last_byte && is_octal_digit(p[1]))                                                                                                                        
 146             ch = ch * 8 + *++p - '0';      // now points at last digit    
```
This second test is the only one that would fail on the following off-by-one error:
```
 -144           if (p < last_byte && is_octal_digit(p[1])) ch = ch * 8 + *++p - '0';
 +144           if (p < last_byte && is_octal_digit(p[2])) ch = ch * 8 + *++p - '0';                                          
 145           if (p < last_byte && is_octal_digit(p[1]))                                                                                                                        
 146             ch = ch * 8 + *++p - '0';      // now points at last digit    
```

Reply:
Oh, I see what you're after, thanks.  You can actually trigger that second case without specifying the size of the string_view:

```
TEST_F(CUnescapeTest, UnescapeOctal) {
  std::string original_string = "\\1z3";
  EXPECT_TRUE(absl::CUnescape(original_string, &result_string_));
  EXPECT_EQ("\1z3", result_string_);
}
```

Let's do that.  It keeps the test minimal.  By specifying the size, I assumed that's what was being tested.  Then we can rename `NonNullTerminatedOctal2` to `Octal` and rename `NonNullTerminatedOctal1` to `NonNullTerminatedOctal`, and then everything looks good to me.

Thanks for bearing with me on this, and sorry about this slow review.